pkg/backend/display/progress.go

```diff
@@ -274,6 +274,61 @@ func ShowProgressEvents(op string, action apitype.UpdateKind, stack tokens.Stack
 	close(done)
 }
 
+// RenderProgressEvents renders the engine events as if to a terminal, providing a simple interface
+// for rendering the progress of an update.
+//
+// A "simple" terminal is used which does not render control sequences. The simple terminal's output
+// is written to opts.Stdout.
+//
+// For consistent output approximating a terminal, consider setting:
+//
+//	opts.Color = colors.Never
+//	opts.RenderOnDirty = false
+//	opts.ShowResourceChanges = true
+//	opts.IsInteractive = true
```

Comment:
should we force some or all of these options? without them, we'll still get a series of frames, yes?

Reply:
We could, think we ought to? It does simplify the contract with the caller if we set color, render on dirty, and is interactive.

`ShowResourceChanges` was added to the list inadvertently, it isn't a prerequisite.

Reply:
Updated